plugins/inputs/snmp/README.md

```diff
@@ -201,6 +201,113 @@ One [metric][] is created for each row of the SNMP table.
+which value is IfIndex from ifTable
+ * Adva entityFacilityTable with translation field: `ADVA-FSPR7-MIB::entityFacilityOneIndex`,
+which value is IfIndex from ifTable
+ * Cisco cpeExtPsePortTable with translation field: `CISCO-POWER-ETHERNET-EXT-MIB::cpeExtPsePortEntPhyIndex`,
+which value is index from entPhysicalTable
+   
+Such field can be used to translate index to secondary table with `secondary_index_table = true`
+and all fields from secondary table (with index pointed from translation field), should have added option
+`secondary_index_use = true`. Telegraf cannot duplicate entries during join so translation
+must be 1-to-1 (not 1-to-many). To add fields from secondary table with index that is not present
+in translation table (outer join), there is a second option for translation index `secondary_outer_join = true`.
+
+###### Example configuration for table joins
+
+CISCO-POWER-ETHERNET-EXT-MIB table before join:
+```
+[[inputs.snmp.table]]
+name = "ciscoPower"
+index_as_tag = true
+
+[[inputs.snmp.table.field]]
+name = "PortPwrConsumption"
+oid = "CISCO-POWER-ETHERNET-EXT-MIB::cpeExtPsePortPwrConsumption"
+
+[[inputs.snmp.table.field]]
+name = "EntPhyIndex"
+oid = "CISCO-POWER-ETHERNET-EXT-MIB::cpeExtPsePortEntPhyIndex"
+```
+
+Partial result (removed agent_host and host columns from all following outputs in this section):
+```
+> ciscoPower,index=1.2 EntPhyIndex=1002i,PortPwrConsumption=6643i 1621460628000000000
+> ciscoPower,index=1.6 EntPhyIndex=1006i,PortPwrConsumption=10287i 1621460628000000000
+> ciscoPower,index=1.5 EntPhyIndex=1005i,PortPwrConsumption=8358i 1621460628000000000
+```
+
+Note here that EntPhyIndex column carries index from ENTITY-MIB table, config for it:
+```
+[[inputs.snmp.table]]
+name = "entityTable"
+index_as_tag = true
+
+[[inputs.snmp.table.field]]
+name = "EntPhysicalName"
+oid = "ENTITY-MIB::entPhysicalName"
+```
+Partial result:
+```
+> entityTable,index=1006 EntPhysicalName="GigabitEthernet1/6" 1621460809000000000
+> entityTable,index=1002 EntPhysicalName="GigabitEthernet1/2" 1621460809000000000
+> entityTable,index=1005 EntPhysicalName="GigabitEthernet1/5" 1621460809000000000
+```
+
+Now, lets attempt to join these results into one table. EntPhyIndex matches index
+from second table, and lets convert EntPhysicalName into tag, so second table will
+only provide tags into result. Configuration:
+
+```
+[[inputs.snmp.table]]
+name = "ciscoPowerEntity"
+index_as_tag = true
+
+[[inputs.snmp.table.field]]
+name = "PortPwrConsumption"
+oid = "CISCO-POWER-ETHERNET-EXT-MIB::cpeExtPsePortPwrConsumption"
+
+[[inputs.snmp.table.field]]
+name = "EntPhyIndex"
+oid = "CISCO-POWER-ETHERNET-EXT-MIB::cpeExtPsePortEntPhyIndex"
+secondary_index_table = true    # enables joining
+
+[[inputs.snmp.table.field]]
+name = "EntPhysicalName"
+oid = "ENTITY-MIB::entPhysicalName"
+secondary_index_use = true      # this tag is indexed from secondary table
+is_tag = true
```

Comment:
does it make sense to reference the table by name instead of just enabling this via booleans?

Reply:
Oh yeah, that also solves the issue if you have multiple of that kind of tables on a host.

`secondary_index_table = "entityTable"`

I love that idea. What do you think @helotpl?

Reply:
lines 294:303 are `[[inputs.snmp.table.fields]]` that are inside single named `[[inputs.snmp.table]]`, there is no way to mix two joins between different `[[inputs.snmp.table]]`, this gives no advantage. We could try to talk about two joins inside single `[[inputs.snmp.table]]`, but this requires rework for this patch not just altering configuration. If you really want this, then perhaps someone else can work on this.

Reply:
I still think you're not understanding what I mean, ignore it.

Reply:
In that case could you present full configuration for quoted example with your suggestions applied?